apps/emqx_resource/src/emqx_resource_health_check.erl

```diff
@@ -54,13 +54,17 @@ delete_checker(Name) ->
         Error -> Error
 	end.
 
-health_check(Name, SleepTime) ->
-    case emqx_resource:health_check(Name) of
-        ok ->
-            emqx_alarm:deactivate(Name);
-        {error, _} ->
-            emqx_alarm:activate(Name, #{name => Name},
-                <<Name/binary, " health check failed">>)
+health_check(Name, SleepTime, Timeout) ->
+    try
```

Comment:
We can move this try...catch into the `emqx_resource:health_check/2`, and make it return `{error, {health_check, timeout}}` in case of timeout.